hypothesis-python/src/hypothesis/stateful.py

```diff
@@ -117,9 +117,12 @@ def run_state_machine(factory, data):
 
             while should_continue.more():
                 value = data.conjecture_data.draw(machine.steps())
-                if print_steps:
-                    machine.print_step(value)
-                machine.execute_step(value)
+                result = None
```

Comment:
It would be handy to include a comment here explaining that this value is only used by print_step if the step has target bundles - otherwise later readers will have to go do the same code-review I just went through 😉

Reply:
I added the comment next to the invocation of 'print_step' below; is this what you meant?